apps/extension/src/state/tx-approval.ts

```diff
@@ -78,6 +79,15 @@ export const createTxApprovalSlice = (): SliceCreator<TxApprovalSlice> => (set,
 
       state.txApproval.choice = undefined;
     });
+
+    window.onbeforeunload = get().txApproval.handleCloseWindow;
+  },
+
+  handleCloseWindow: () => {
+    set(state => {
+      state.txApproval.choice = UserChoice.Ignored;
+    });
+    get().txApproval.sendResponse();
```

Comment:
ah, this is incorrect. ignored is for use by origin approval to indicate the source of a request should be ignored, and not able to trigger any more requests.

in this context it's meaningless. without this empty sendResponse, would get a chrome.runtime error at the caller, indicating the window closed.

Reply:
OK, I'll fix this